qiskit/transpiler/basepasses.py

```diff
@@ -22,19 +22,16 @@ class MetaPass(type):
     def __call__(cls, *args, **kwargs):
         if '_pass_cache' not in cls.__dict__.keys():
             cls._pass_cache = {}
-        args, kwargs = cls.normalize_parameters(*args, **kwargs)
-        hash_ = hash(MetaPass._freeze_init_parameters(cls.__init__, args, kwargs))
-        if hash_ not in cls._pass_cache:
-            new_pass = type.__call__(cls, *args, **kwargs)
-            cls._pass_cache[hash_] = new_pass
-        return cls._pass_cache[hash_]
+        pass_instance = type.__call__(cls, *args, **kwargs)
+        pass_instance._hash = hash(MetaPass._freeze_init_parameters(cls, args, kwargs))
+        return pass_instance
 
     @staticmethod
-    def _freeze_init_parameters(init_method, args, kwargs):
+    def _freeze_init_parameters(class_, args, kwargs):
         self_guard = object()
-        init_signature = signature(init_method)
+        init_signature = signature(class_.__init__)
         bound_signature = init_signature.bind(self_guard, *args, **kwargs)
-        arguments = []
+        arguments = [('class_.__name__', class_.__name__)]
```

Comment:
Why do you need this?

Reply:
Because the name is also part of the identification, no just the arguments.